### ISSUE TITLE
DELETE request will delete all data

### ISSUE THREAD

**Issue by MiYogurt** [typicode/json-server#885]
Version: 0.14.0

```json
{
  "layers": [
    {
      "id": "7d-UXfgmm",
      "componentId": "aaa"
    },
    {
      "id": "HXf8PkKl7",
      "componentId": "aaa"
    },
    {
      "id": "kNJ083om1",
      "componentId": "aaa"
    },
    {
      "id": "DYPg3hl5s",
      "componentId": "aaa"
    }
  ],
  "components": [
    {
      "id": "K8iv0hJaQ",
      "label": "component"
    }
  ]
}
```

```bash
http DELETE :3000/layers/HXf8PkKl7
```

The will delete all layers 。

**Comment by Akiyamka** [comment]
Still not fixed...

**Comment by ayxos** [comment]
same here

**Comment by faradayfan** [comment]
I'm having this problem as well.

**Comment by Zhouqchao** [comment]
This bug still happens today!

**Comment by yassotreyo** [comment]
same here

**Comment by Damoness** [comment]
I'm having the some problem too .

**Comment by zhangweiHello** [comment]
I think a simple framework implements a simple problem，  why not use [aok.js](https://github.com/apporoad/aok.js)

**Comment by acuriel** [comment]
I'm having the same issues...multiple instances of the same element randomly deleted when DELETE request to one single element

**Comment by MicroRobi** [comment]
Hi!
This is not a random error.
I found that if there is some kind of connection between elements, then DELETE request will erase both of them:
I have a "database":

```
{
    "users":[
        {"id":1, "name": "Bob", "email":"bob@gmailcom", "age": 35},
        {"id":2, "name": "Sue", "email":"sue@gmailcom", "age": 32},
        {"id":3, "name": "Cloe", "email":"cloe@gmailcom", "age": 19},
        {"id":4, "name": "Jane", "email":"jane@gmailcom", "age": 26}
    ],
    "comments":[
        {"id":1, "message1": "Here should be some comment", "userId": 2},
        {"id":2, "message2": "Here should be many comment", "userId": 4},
        {"id":3, "message3": "Here should be more comment", "userId": 1}
    ]
}
```
and I run this from console: 
`let fetchOpt = {
    method: "delete",
    mode: "cors",
    cache: "no-cache"
};
fetch("http://localhost:3000/users/4", fetchOpt);`

then server will delete "Jane with **id:4**" from users, and "message2 with **userId: 4**" from comments

Try it. Then I changed "userId" to "usr" and there is no more "strange delete"

**Comment by Akiyamka** [comment]
Expected behaviour:
https://github.com/typicode/json-server/blob/master/__tests__/server/plural-with-custom-foreign-key.js#L110

default value: 
https://github.com/typicode/json-server/blob/574cb817dc0bb2a67cd5ed38a0fc3211d4058804/src/server/router/index.js#L16

logic:
https://github.com/typicode/json-server/blob/7ce599c50290e5c3a2336fd5b0bd84c3508df236/src/server/router/plural.js#L329

**Comment by vinicamposdev** [comment]
I'm having this issue with version 0.16.3

**Comment by antonreshetov** [comment]
same issue

**Comment by wll8** [comment]
I debugged the source code and found that the logic is: **delete all data that are not related to each other (through xxId as the association flag).**

When we delete data, it is equivalent to cleaning the entire db.json once, regardless of whether it is the currently operating table.

Here are some relevant test cases: [plural.js#L628](https://github.com/typicode/json-server/blob/master/__tests__/server/plural.js#L628)

![image](https://user-images.githubusercontent.com/28223260/200783252-567646c6-74bc-497b-8a80-115f2afe98d7.png)

**Comment by loosq** [comment]
If you are trying to delete aome object with unexisting foreign key, the delete request will delete all data

**Comment by wll8** [comment]
This is the solution: https://github.com/wll8/json-server#options_noremovedependents

**Comment by nicotakenotice** [comment]
TL;DR
Run `json-server` with the flag `--foreignKeySuffix SomeValue`.

`SomeValue` can be whatever you want, it just needs to be different from the default [`Id`](https://github.com/typicode/json-server#cli-usage) and, of course, from whatever suffix you usually write when estabilishing a relationship between objects.

Overall, @wll8 solution is the best to handle this issue.

**Comment by xuananh2212** [comment]
how to delete all ?
